File: fsharp/FsFind/FsFindTests/FindFileTests.fs
namespace FsFindTests

open System.IO
open NUnit.Framework
open FsFind

[<TestFixture>]
type FindFileTests () =

    member this.CsFindPath = "~/src/xfind/csharp/CsFind/CsFind"
    member this.WinCsFindPath = @"C:\src\xfind\csharp\CsFind\CsFind"

    [<SetUp>]
    member this.Setup () =
        ()

    [<Test>]
    member this.FindFile_ToString_EqualsExpected () =
        let findFile = FindFile.Create (FileInfo(FileUtil.JoinPath this.CsFindPath "Finder.cs")) FileType.Code
        Assert.AreEqual(this.CsFindPath + "/Finder.cs", FindFile.ToString(findFile))
        ()

    [<Test>]
    member this.FindFileTrailingSlash_ToString_EqualsExpected () =
        let findFile = FindFile.Create (FileInfo(FileUtil.JoinPath this.CsFindPath "Finder.cs")) FileType.Code
        Assert.AreEqual(this.CsFindPath + "/Finder.cs", FindFile.ToString(findFile))
        ()

    [<Test>]
    member this.FindFileBackSlashes_ToString_EqualsExpected () =
        let findFile = FindFile.Create (FileInfo(FileUtil.JoinPath this.WinCsFindPath "Finder.cs")) FileType.Code
        Assert.AreEqual(this.WinCsFindPath + @"\Finder.cs", FindFile.ToString(findFile))
        ()

    [<Test>]
    member this.FindFileBackSlashesTrailingSlash_ToString_EqualsExpected () =
        let findFile = FindFile.Create (FileInfo(FileUtil.JoinPath this.WinCsFindPath "Finder.cs")) FileType.Code
        Assert.AreEqual(this.WinCsFindPath + @"\Finder.cs", FindFile.ToString(findFile))
        ()

File: csharp/CsFind/CsFindTests/FindFileTests.cs
using NUnit.Framework;
using CsFindLib;

namespace CsFindTests
{
	[TestFixture]
	class FindFileTests
	{
		private const string CsFindPath = "~/src/xfind/csharp/CsFind/CsFind";
		private const string WinCsFindPath = @"C:\src\git\xfind\csharp\CsFind\CsFind";

		[Test]
		public void FindFile_ToString_EqualsExpected()
		{
			var findFile = new FindFile(CsFindPath, "Finder.cs", FileType.Text);
			Assert.AreEqual(CsFindPath + "/Finder.cs", findFile.ToString());
		}

		[Test]
		public void FindFileTrailingSlash_ToString_EqualsExpected()
		{
			var findFile = new FindFile(CsFindPath + "/", "Finder.cs", FileType.Text);
			Assert.AreEqual(CsFindPath + "/Finder.cs", findFile.ToString());
		}

		[Test]
		public void FindFileBackSlashes_ToString_EqualsExpected()
		{
			var findFile = new FindFile(WinCsFindPath, "Finder.cs", FileType.Text);
			Assert.AreEqual(WinCsFindPath + @"\Finder.cs", findFile.ToString());
		}

		[Test]
		public void FindFileBackSlashesTrailingSlash_ToString_EqualsExpected()
		{
			var findFile = new FindFile(WinCsFindPath + @"\", "Finder.cs", FileType.Text);
			Assert.AreEqual(WinCsFindPath + @"\Finder.cs", findFile.ToString());
		}
	}
}
